kore/src/Kore/Step/Simplification/SubstitutionSimplifier.hs

```diff
@@ -383,7 +383,7 @@ simplifySubstitutionWorker sideCondition makeAnd' = \substitution -> do
         addPredicate predicate
         return substitution'
 
-    sideConditionRepresentation = SideCondition.toRepresentation sideCondition
+    ~sideConditionRepresentation = SideCondition.toRepresentation sideCondition
```

Comment:
Does this still need to be explicitly lazy, with `SideCondition.Representation` itself being lazy?

Reply:
`SideCondition.toRepresentation` also [maps](https://github.com/kframework/kore/blob/4195d948dbee2fddd9f056f7bb1d26935e384c6f/kore/src/Kore/Internal/SideCondition.hs#L386) the variables of the `SideCondition` it receives.